### PR TITLE
Batch tensor conversions to minimize the RPC count, compile, execute,…

### PR DESCRIPTION
**Pull request by dlibenzi**
… times.

### CODE REVIEW

File: torch_xla/csrc/tensor.h [hunk 1]
```diff
@@ -174,8 +174,12 @@ class XLATensor {
   static void ApplyPendingGraph(std::vector<XLATensor>* tensors,
                                 ApplyContext* apply_context);
 
-  // Retrieves the PyTorch tensors behind the XLA tensors.
-  static std::vector<at::Tensor> GetTensors(std::vector<XLATensor>* tensors);
+  // Retrieves the PyTorch tensors behind the XLA tensors. If the writeable
+  // vector is not nullptr, it must be teh same size as tensors, and the
```
Comment (asuhan) [inline torch_xla/csrc/tensor.h]: teh -> the

File: torch_xla/csrc/aten_xla_bridge.h [hunk 2]
```diff
@@ -21,7 +21,11 @@ at::Tensor CreateEmptyTensor(at::IntList size,
 XLATensor& GetXlaTensor(const at::Tensor& tensor);
 
 // Creates a vector of at::Tensor objects extracted from a list of XLA tensors.
-std::vector<at::Tensor> XlaCreateTensorList(const at::TensorList& tensors);
+// If the writeable vector is not nullptr, it must be teh same size as tensors,
```
Comment (asuhan) [inline torch_xla/csrc/aten_xla_bridge.h]: teh -> the

File: scripts/gen.py [hunk 3]
```diff
@@ -264,6 +264,39 @@ def append(self, s):
...
+    self.tensors.append(name)
+    self.writeable.append(writeable)
+    return '{}[{}]'.format(self.var_name, len(self.tensors) - 1)
+
+  def generate(self):
+    tvar_name = '{}_tensors'.format(self.var_name)
+    wvar_name = '{}_writeables'.format(self.var_name)
+    code = ''
+    code += '  std::vector<at::Tensor> {} = {{'.format(tvar_name)
+    for i, t in enumerate(self.tensors):
```
Comment (asuhan) [inline scripts/gen.py]: Use `', '.join(self.tensors)`.

File: scripts/gen.py [hunk 4]
```diff
@@ -264,6 +264,39 @@ def append(self, s):
...
+      if i > 0:
+        code += ', '
+      code += t
+    code += '};\n'
+    code += '  std::vector<bool> {} = {{'.format(wvar_name)
+    for i, w in enumerate(self.writeable):
```
Comment (asuhan) [inline scripts/gen.py]: Here you can use string join too.

File: torch_xla/csrc/aten_xla_bridge.h (hunk 2 above)
Comment (dlibenzi) [reply torch_xla/csrc/aten_xla_bridge.h]: Done

File: torch_xla/csrc/tensor.h (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/tensor.h]: Done

File: scripts/gen.py (hunk 3 above)
Comment (dlibenzi) [reply scripts/gen.py]: Done

File: scripts/gen.py (hunk 4 above)
Comment (dlibenzi) [reply scripts/gen.py]: Yup

**Pull request merged**